controllers/nodehealthcheck_controller.go

```diff
@@ -388,14 +393,27 @@ func (r *NodeHealthCheckReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	return result, nil
 }
 
-func (r *NodeHealthCheckReconciler) isClusterUpgrading() bool {
+func (r *NodeHealthCheckReconciler) isClusterUpgrading(nodesToBeRemediated []v1.Node) bool {
 	clusterUpgrading, err := r.ClusterUpgradeStatusChecker.Check()
 	if err != nil {
 		// if we can't reliably tell if the cluster is upgrading then just continue with remediation.
 		// TODO finer error handling may help to decide otherwise here.
 		r.Log.Error(err, "failed to check if the cluster is upgrading. Proceed with remediation as if it is not upgrading")
 	}
-	return clusterUpgrading
+	return clusterUpgrading || r.isHostedControlPlaneUpgrading(nodesToBeRemediated)
+}
+
+func (r *NodeHealthCheckReconciler) isHostedControlPlaneUpgrading(nodesToBeRemediated []v1.Node) bool {
```

Comment:
- I'm not sure if this check is that specific for HCP, I would prefer a more generic name, e.g. `isNodeUpgrading`, with a comment that it's needed on e.g. HCP clusters
- Is there a reason why you didn't add it to the ClusterUpgradeChecker but here? 🤔

Reply:
>I'm not sure if this check is that specific for HCP, I would prefer a more generic name, e.g. isNodeUpgrading, with a comment that it's needed on e.g. HCP clusters

Sure, sounds good

>Is there a reason why you didn't add it to the ClusterUpgradeChecker but here? 🤔

Mostly because I didn't want to modify the interface and `isClusterUpgrading` seemed like a reasonable fit as a wrapper to use.
By any means it's not something I feel strongly about, so let me know if you prefer it changed.

Reply:
yes I prefer to change it, 

- in order keep cluster upgrade check at one place
- another benefit is that we skip the annotation check on non OCP clusters, as the existing check already does